File: update.py
#!/usr/bin/env python3
from lxml import html
from jinja2 import Template
import requests
import yaml

def loadWins():
    page = requests.get('http://stats.washingtonpost.com/fb/standings.asp') # easiest site to parse
    tree = html.fromstring(page.content)
    teams = tree.xpath('//tr[@class="shsRow0Row"]')
    teams += tree.xpath('//tr[@class="shsRow1Row"]')
    
    stats = {}
    for t in teams:
        teamName = str(t[0][0].text_content())
        wins = int(t[1].text_content())
        stats[teamName] = wins

    if len(stats) != 32:
        print('parsing failed') # make into better error

    return stats

def loadTeams():
    with open('teams.yaml', 'r') as fp:
        teams = yaml.load(fp)
    return teams 

def render(score, teams, stats):
    with open('index.html.jinja') as fp:
        tmpl = Template(fp.read())
    # Reformat Score to be template friendly
    score = [{'wins':w,'part':p} for w,p in sorted([(w,p) for p,w in score.items()], reverse=True)]
    with open('index.html','w') as outFile:
        outFile.write(tmpl.render(score=score))
    return

if __name__ == '__main__':
    teams = loadTeams()
    stats = loadWins()
    score = {}
    for part in teams:
        score[part] = 0
        for pro in teams[part]:
            score[part] += stats[pro]
    render(score,teams,stats)

File: update.js
function loadUser() {
  var template = $('#template').html();
  Mustache.parse(template);   // optional, speeds up future uses
  var rendered = Mustache.render(template, {name: "Luke"});
  $('#target').html(rendered);
}

function loadTable() {
    var template = $('#template').html();
    Mustache.parse(template);   // optional, speeds up future uses
    updateScore( function(scores) { 
        var rendered = Mustache.render(template, {score:scores});
        $('#target').html(rendered);
    });
    //var rendered = Mustache.render(template, {score:score});
    //$('#target').html(rendered);
}

function updateScore(callback) {
    loadTeams(function(teams) {
        loadStats(function(stats) {
            var score = [] 
            for (var part in teams) {
                var temp = {'part':part, 'wins':0};
                for (var pro in teams[part]) {
                    temp['wins'] += stats[teams[part][pro]];
                }
                score.push(temp);
            }
            score.sort(function(first,second) {
                return second['wins'] - first['wins'];
            });
            callback(score);
        })
    });
}

function loadTeams(callback) {
    $.getJSON('teams.json', callback);
}

function loadStats(callback) {
    // to be fully implemented
    $.getJSON('stats.json', callback);
}
